backend/data/src/main/kotlin/io/tolgee/service/queryBuilders/translationViewBuilder/CursorPredicateProvider.kt

```diff
@@ -28,33 +28,60 @@ class CursorPredicateProvider(
     var result: Predicate? = null
     cursor?.entries?.reversed()?.forEach { (property, value) ->
       val isUnique = property === KeyWithTranslationsView::keyId.name
-      val expression =
-        selection[property] as? Expression<String>
+      val selected =
+        selection[property]
           ?: throw BadRequestException(Message.CANNOT_SORT_BY_THIS_COLUMN)
 
+      // We need the runtime type of the column
+      val colType = selected.javaType
+      // parse the raw string from cursor into correct type for comparison
+      val typedValue = parseValue(colType, value.value)
+
+      @Suppress("UNCHECKED_CAST")
+      val expression = selected as Expression<Comparable<Any>>
+
       val strongCondition: Predicate
       val condition: Predicate
+
       if (value.direction == Sort.Direction.ASC) {
         condition =
           if (isUnique) {
-            cb.greaterThan(expression, value.value!!)
+            cb.greaterThan(expression, typedValue!!)
           } else {
-            cb.greaterThanOrEqualToNullable(expression, value.value)
+            cb.greaterThanOrEqualToNullable(expression, typedValue)
           }
-        strongCondition = cb.greaterThanNullable(expression, value.value)
+        strongCondition = cb.greaterThanNullable(expression, typedValue)
       } else {
         condition =
           if (isUnique) {
-            cb.lessThan(expression, value.value!!)
+            cb.lessThan(expression, typedValue!!)
           } else {
-            cb.lessThanOrEqualToNullable(expression, value.value)
+            cb.lessThanOrEqualToNullable(expression, typedValue)
           }
-        strongCondition = cb.lessThanNullable(expression, value.value)
+        strongCondition = cb.lessThanNullable(expression, typedValue)
       }
       result = result?.let {
         cb.and(condition, cb.or(strongCondition, result))
       } ?: condition
     }
     return result
   }
+
+  @Suppress("UNCHECKED_CAST")
+  private fun parseValue(
+    javaType: Class<*>,
+    raw: String?,
+  ): Comparable<Any>? {
+    if (raw == null) return null
+
+    return when (javaType) {
```

Comment:
I don't get what this function is supposed to do.

It looks to me like it just converts some types to Comparable, which it already is, right? So what's the point?

Reply:
I think you can just do 

```
raw ?: return null
return rawNotNull as? Comparable<Any> ?: throw IllegalArgumentException("Only comparable values can be used for sorting") 
```